src/RepoDriver.sol

```diff
@@ -0,0 +1,530 @@
+    /// The actual update of the owner will be made in a future transaction.
+    /// The driver will cover the fee in Link that must be paid to the operator.
+    /// If you want to cover the fee yourself, use `onTokenTransfer`.
+    ///
+    /// The repository must contain a `FUNDING.json` file in the project root in the default branch.
+    /// The file must be a valid JSON with arbitrary data, but it must contain the owner address
+    /// as a hexadecimal string under `drips` -> `<CHAIN NAME>` -> `ownedBy`, a minimal example:
+    /// `{ "drips": { "ethereum": { "ownedBy": "0x0123456789abcDEF0123456789abCDef01234567" } } }`.
+    /// If the operator can't read the owner when processing the update request,
+    /// it ignores the request and no change to the user ownership is made.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// For GitHub and GitLab it must follow the `user_name/repository_name` structure
+    /// and it must be formatted identically as in the repository's URL,
+    /// including the case of each letter and special characters being removed.
+    /// @return userId The ID of the user.
+    function requestUpdateOwner(Forge forge, bytes memory name)
+        public
+        whenNotPaused
+        returns (uint256 userId)
+    {
+        uint256 fee = _repoDriverAnyApiStorage().defaultFee;
+        require(linkToken.balanceOf(address(this)) >= fee, "Link balance too low");
+        return _requestUpdateOwner(forge, name, fee);
+    }
+
+    /// @notice The function called when receiving funds from ERC-677 `transferAndCall`.
+    /// Only supports receiving Link tokens, callable only by the Link token smart contract.
+    /// The only supported usage is requesting user ownership updates,
+    /// the transferred tokens are then used for paying the AnyApi operator fee,
+    /// see `requestUpdateOwner` for more details.
+    /// The received tokens are never refunded, so make sure that
+    /// the amount isn't too low to cover the fee, isn't too high and wasteful,
+    /// and the repository's content is valid so its ownership can be verified.
+    /// @param amount The transferred amount, it will be used as the AnyApi operator fee.
+    /// @param data The `transferAndCall` payload.
+    /// It must be a valid ABI-encoded calldata for `requestUpdateOwner`.
+    /// The call parameters will be used the same way as when calling `requestUpdateOwner`,
+    /// to determine which user's ownership update is requested.
+    function onTokenTransfer(address, /* sender */ uint256 amount, bytes calldata data)
+        public
+        whenNotPaused
+    {
+        require(msg.sender == address(linkToken), "Callable only by the Link token");
+        require(data.length >= 4, "Data not a valid calldata");
+        require(bytes4(data[:4]) == this.requestUpdateOwner.selector, "Data not requestUpdateOwner");
```

Comment:
Mhm, as far as I know, the `data` is optional and it is up to us, how we want to define it.
We could just assume it is only the `data` and not the function signature. 
I think if we want this additional safety check, we should call the corresponding function and not the internal one. 
It would also solve the problem of the fee.

Reply:
If we drop the signature, we'll lose any possibility of adding other behaviors on receiving tokens, which makes me slightly uncomfortable. Having a full calldata of an existing public function makes it convenient to build the payload, there are tools both in Solidity and in JS. Passing a full calldata payload is also a normal thing in the ecosystem, e.g. the operator does that.

The fee parameter behavior is intended, we're discussing it above.

Reply:
Yeah, that's right. 
My point was checking for `this.functionA.selector` and then calling `functionB`. However, maybe i should see it more like calling. `function_A` :).

I guess it is okay.